includes/html/table/arp-search.inc.php

```diff
@@ -1,121 +1,104 @@
 <?php
 
+use App\Models\Ipv4Mac;
 use LibreNMS\Util\Mac;
+use LibreNMS\Util\Rewrite;
+use LibreNMS\Util\Url;
 
 $param = [];
 
-$sql = ' FROM `ipv4_mac` AS M, `ports` AS P, `devices` AS D ';
-$where = '';
+if (! isset($sort) || empty($sort)) {
+    $sort = 'hostname ASC';
+}
+$sort_arr = explode(' ', trim($sort));
+if ($sort_arr[0] === 'interface') {
+    $sort_arr[0] = 'port_if_descr';
+} elseif ($sort_arr[0] === 'hostname') {
+    $sort_arr[0] = 'device_hostname';
+}
 
-if (! Auth::user()->hasGlobalRead()) {
-    $device_ids = Permissions::devicesForUser()->toArray() ?: [0];
-    $where .= ' AND `D`.`device_id` IN ' . dbGenPlaceholders(count($device_ids));
-    $param = array_merge($param, $device_ids);
+if (isset($current)) {
+    $page = $current;
+} else {
+    $page = 1;
 }
 
-$sql .= " WHERE M.port_id = P.port_id AND P.device_id = D.device_id $where ";
+$user = Auth::user();
+$query = Ipv4Mac::with('port', 'device', 'remote_ports_maybe', 'remote_ports_maybe.device')
+    ->withAggregate('port', 'ifDescr')
+    ->withAggregate('device', 'hostname')
+    ->orderBy(...$sort_arr);
+
+if (! $user->hasGlobalRead()) {
```

Comment:
Might be better to use the hasAccess scope

Reply:
Done